apps/edxapp/vars/vault/main.yml.j2

```
@@ -0,0 +1,36 @@
+# secret: edxapp
+# customer: {{ customer }}
+# env_type: {{ env_type }}
+
+# Mysql
+{% set mysql_credentials = mysql | json_query("[?release=='" ~ edxapp_mysql_version ~ "'].databases | [0][?application=='edxapp'].{user: user, password: password, name: name} | [0]") %}
+MYSQL_ROOT_PASSWORD: ""
+MYSQL_ALLOW_EMPTY_PASSWORD: yes
+MYSQL_DATABASE: {{ mysql_credentials.name }}
+MYSQL_USER: {{ mysql_credentials.user }}
+MYSQL_PASSWORD: {{ mysql_credentials.password }}
+
+# MongoDB
+{% set mongodb_credentials = mongodb | json_query("[?release=='" ~ edxapp_mongodb_version ~ "'].databases | [0][?application=='edxapp'].{user: user, password: password, name: name} | [0]") %}
+MONGODB_ADMIN_PASSWORD: password
+MONGODB_USER: {{ mongodb_credentials.user }}
+MONGODB_PASSWORD: {{ mongodb_credentials.password }}
+MONGODB_DATABASE: {{ mongodb_credentials.name }}
```

Have you tried to make those variables `mandatory`?

```suggestion
MONGODB_DATABASE: {{ mongodb_credentials.name | mandatory }}
```

To ensure that if the corresponding variable is not defined in the databases vault, this templates fails to compile.

---

I tried, it fails but the output error does not give information about which variable is mandatory. Not mentionning it as mandatory give the information about which variable is missing in the output error. I suggest to not use the mandatory filter

---

Ok then.